jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -545,6 +558,29 @@ public boolean handle()
         return !suspended;
     }
 
+    private void ensureConnectionClose()
+    {
+        _response.getHttpFields().computeField(HttpHeader.CONNECTION, (h, fields) ->
+        {
+            if (fields == null || fields.isEmpty())
+                return HttpConnection.CONNECTION_CLOSE;
+
+            if (fields.stream().anyMatch(f -> f.contains(HttpHeaderValue.CLOSE.asString())))
+            {
+                if (fields.size() == 1)
+                    return fields.get(0);
+
+                return new HttpField(HttpHeader.CONNECTION, fields.stream()
+                    .flatMap(field -> Stream.of(field.getValues()))
+                    .collect(Collectors.joining(", ")));
+            }
+
+            return new HttpField(HttpHeader.CONNECTION, fields.stream()
+                .flatMap(field -> Stream.of(field.getValues()))
+                .collect(Collectors.joining(", ")) + ",close");
```

Comment:
You are using ", " (comma+space) to delimit value, so add a space before "close" when you add it.
Same as above, should duplicates be discarded?